### PR TITLE
Implement future and future-call

### PR DESCRIPTION
**Pull request by patrickgombert**
Also adds timeout arity to deref

### CODE REVIEW

**Comment by patrickgombert** [comment]
```
Build Matrix
Job	Duration	Finished	JDK	OS
 234.1	23 sec	3 minutes ago	openjdk6	linux - PASS
 234.2	39 sec	less than a minute ago	openjdk7	linux - FAIL
 234.3	38 sec	less than a minute ago	oraclejdk7	linux - PASS
```

Something weird (a bug maybe?) is happening with `java.util.concurrent.Future` on openjdk7. I'll dig in tonight.

**Comment by patrickgombert** [comment]
Had to use a hack, I'm going to keep trying to make this work without the hack, but it is mergable.

File: project.clj [hunk 1]
```diff
@@ -4,7 +4,8 @@
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
 
-  :dependencies [[org.clojure/clojure "1.6.0"]]
+  :dependencies [[org.clojure/clojure "1.6.0"]
+                 [org.clojure/tools.emitter.jvm "0.1.0-beta5"]]
```
Comment (mylesmegyesi) [inline project.clj]: Is this used?

Comment (patrickgombert) [inline project.clj]: no, it was for debugging

removed

**Pull request merged**